lib/service.js

```diff
@@ -349,12 +367,13 @@ AWS.Service = inherit({
   attemptFailEvent: function attemptFailEvent(request) {
     var monitoringEvent = this.apiAttemptEvent(request);
     var response = request.response;
+    var error = response.error;
     if (response.httpResponse.statusCode > 299 ) {
```

Comment:
Just echoing diff at line 309, is `statusCode &&` check also needed here?

Reply:
`statusCode` always exists so I remove the redundant check at line 309